File: core/cbt_library.py
"""
CBT & Clinical Intervention Library for Smart Health Risk Indicator (SHRI)
"""

CBT_LIBRARY = {
    "depression": [
        {
            "id": "cbt-dep-1",
            "title": "Behavioral Activation & Micro-Scheduling",
            "evidence_grade": "Grade A (NICE Guidelines)",
            "description": "Schedule 1 small, pleasurable mastery activity daily (e.g. 15-minute walk, drawing) to break depressive inertia and restore dopamine feedback loops.",
            "target_domain": "Depression / Anhedonia",
            "milestone": "Complete 5 scheduled activities in 7 days"
        },
        {
            "id": "cbt-dep-2",
            "title": "Cognitive Restructuring (3C Model)",
            "evidence_grade": "Grade A (Beck Institute)",
            "description": "Catch negative automatic thoughts ('I fail at everything'), Check the objective evidence, and Change to a balanced realistic alternative.",
            "target_domain": "Depression / Cognitive Distortions",
            "milestone": "Log 3 automatic thought records per week"
        },
        {
            "id": "cbt-dep-3",
            "title": "Gratitude & Positive Savoring Journaling",
            "evidence_grade": "Grade B (Empirically Supported)",
            "description": "Write down 3 specific things that went well each evening before bed to retrain attentional bias away from depressive rumination.",
            "target_domain": "Mood / Rumination",
            "milestone": "Maintain 10 consecutive daily logs"
        }
    ],
    "anxiety": [
        {
            "id": "cbt-anx-1",
            "title": "Physiological Grounding & 4-7-8 Diaphragmatic Breathing",
            "evidence_grade": "Grade A (Physiological Reg)",
            "description": "Inhale 4s through nose, hold 7s, exhale 8s with pursed lips. Stimulates the vagus nerve and activates the parasympathetic relaxation response.",
            "target_domain": "Anxiety / Somatic Hyperarousal",
            "milestone": "Practice 2 sessions daily for 14 days"
        },
        {
            "id": "cbt-anx-2",
            "title": "Graded Exposure Hierarchy (Systematic Desensitization)",
            "evidence_grade": "Grade A (Gold Standard)",
            "description": "Build a 1–10 fear ladder for anxiety triggers (e.g. social interactions, speaking up) and systematically expose in safe micro-steps.",
            "target_domain": "Generalized & Social Anxiety",
            "milestone": "Conquer 2 rungs on fear hierarchy within 3 weeks"
        },
        {
            "id": "cbt-anx-3",
            "title": "Worry Time Postponement Technique",
            "evidence_grade": "Grade B (CBT Standard)",
            "description": "Designate a fixed 15-minute daily 'Worry Window'. Whenever anxious intrusive thoughts arise during the day, jot them down and postpone review.",
            "target_domain": "Cognitive Worry / Overthinking",
            "milestone": "Postpone daily intrusive rumination 5 times"
        }
    ],
    "sleep": [
        {
            "id": "cbt-slp-1",
            "title": "Stimulus Control Therapy (Bed = Sleep Only)",
            "evidence_grade": "Grade A (AASM Guidelines)",
            "description": "Remove smartphones/screens from bed. If awake after 20 minutes, get out of bed and engage in a dim-light non-stimulating activity until drowsy.",
            "target_domain": "Insomnia / Sleep Latency",
            "milestone": "Achieve zero screen usage in bed for 10 days"
        },
        {
            "id": "cbt-slp-2",
            "title": "Circadian Anchoring & Sleep Restriction",
            "evidence_grade": "Grade A (Sleep Medicine)",
            "description": "Maintain a strict, identical wake-up time 7 days a week, regardless of total hours slept, to stabilize circadian melatonin oscillation.",
            "target_domain": "Circadian Rhythm / Sleep Quality",
            "milestone": "Consistent wake-time within ±15 minutes for 2 weeks"
        }
    ],
    "lifestyle": [
        {
            "id": "cbt-lif-1",
            "title": "Digital Sunsetting & Blue-Light Reduction",
            "evidence_grade": "Grade B (Digital Hygiene)",
            "description": "Establish a digital curfew 60 minutes prior to bedtime and limit recreational screen time to under 3.5 hours/day.",
            "target_domain": "Digital Overload / Sleep Quality",
            "milestone": "Reduce average screen time by 1.5 hours daily"
        },
        {
            "id": "cbt-lif-2",
            "title": "Aerobic Micro-Burst Protocol",
            "evidence_grade": "Grade A (Neurogenesis & BDNF)",
            "description": "Engage in 25–30 minutes of moderate cardiovascular movement (cycling, brisk walking, sports) 4 times a week.",
            "target_domain": "Physical Activity / Somatic Vigor",
            "milestone": "Log 120 minutes of moderate activity weekly"
        }
    ]
}


def get_recommendations(score, domain_profile):
    """
    Selects evidence-based CBT interventions based on composite risk score and dominant symptom domain.
    """
    recs = []
    depr_s = domain_profile.get("Depression", 0)
    anx_s = domain_profile.get("Anxiety", 0)
    slp_s = domain_profile.get("Sleep Disruption", 0)

    if depr_s >= anx_s and depr_s >= 30:
        recs.extend(CBT_LIBRARY["depression"])
    elif anx_s > depr_s and anx_s >= 30:
        recs.extend(CBT_LIBRARY["anxiety"])
    else:
        recs.extend(CBT_LIBRARY["anxiety"][:1] + CBT_LIBRARY["depression"][:1])

    if slp_s >= 35 or domain_profile.get("Lifestyle & Behavioral Risk", 0) >= 40:
        recs.extend(CBT_LIBRARY["sleep"])
        recs.extend(CBT_LIBRARY["lifestyle"][:1])

    return recs[:4]


def classify_risk(score):
    """
    Maps continuous 0-100 risk score to 5 clinical risk tiers.
    """
    if score < 20:
        return {"label": "Minimal Risk", "tier": 1, "color": "#10B981", "badge": "success", "description": "Symptoms within healthy normative bounds. Continued wellness monitoring."}
    elif score < 40:
        return {"label": "Mild Risk", "tier": 2, "color": "#3B82F6", "badge": "info", "description": "Mild sub-clinical symptoms. Digital psychoeducation and lifestyle self-care recommended."}
    elif score < 60:
        return {"label": "Moderate Risk", "tier": 3, "color": "#F59E0B", "badge": "warning", "description": "Moderate symptoms. Active clinician check-in and targeted CBT protocol recommended."}
    elif score < 80:
        return {"label": "High Risk", "tier": 4, "color": "#F97316", "badge": "warning", "description": "High clinical risk. Prioritize clinical safety review and formal psychiatric assessment."}
    else:
        return {"label": "Severe Risk", "tier": 5, "color": "#EF4444", "badge": "danger", "description": "Critical clinical severity. Immediate clinical escalation and emergency safety planning required."}

File: js/cbt_library.js
/**
 * CBT Interventions & FHIR Export Engine (Client-Side JS)
 */

const CBT_DATA = {
    depression: [
        {
            title: "Behavioral Activation & Micro-Scheduling",
            grade: "Grade A (NICE Guidelines)",
            desc: "Schedule 1 small, pleasurable mastery activity daily (15-min walk, drawing) to break depressive inertia and restore dopamine feedback loops.",
            domain: "Depression / Anhedonia",
            milestone: "Complete 5 scheduled activities in 7 days"
        },
        {
            title: "Cognitive Restructuring (3C Model)",
            grade: "Grade A (Beck Institute)",
            desc: "Catch negative automatic thoughts ('I fail at everything'), Check the objective evidence, and Change to a balanced realistic alternative.",
            domain: "Depression / Cognitive Distortions",
            milestone: "Log 3 automatic thought records per week"
        }
    ],
    anxiety: [
        {
            title: "Physiological Grounding & 4-7-8 Diaphragmatic Breathing",
            grade: "Grade A (Physiological Reg)",
            desc: "Inhale 4s through nose, hold 7s, exhale 8s with pursed lips. Stimulates the vagus nerve and activates the parasympathetic relaxation response.",
            domain: "Anxiety / Somatic Hyperarousal",
            milestone: "Practice 2 sessions daily for 14 days"
        },
        {
            title: "Graded Exposure Hierarchy (Systematic Desensitization)",
            grade: "Grade A (Gold Standard)",
            desc: "Build a 1–10 fear ladder for anxiety triggers (social interactions, speaking up) and systematically expose in safe micro-steps.",
            domain: "Generalized & Social Anxiety",
            milestone: "Conquer 2 rungs on fear hierarchy within 3 weeks"
        }
    ],
    sleep: [
        {
            title: "Stimulus Control Therapy (Bed = Sleep Only)",
            grade: "Grade A (AASM Guidelines)",
            desc: "Remove smartphones/screens from bed. If awake after 20 minutes, get out of bed and engage in a dim-light non-stimulating activity until drowsy.",
            domain: "Insomnia / Sleep Latency",
            milestone: "Achieve zero screen usage in bed for 10 days"
        }
    ],
    lifestyle: [
        {
            title: "Digital Sunsetting & Blue-Light Curfew",
            grade: "Grade B (Digital Hygiene)",
            desc: "Establish a digital curfew 60 minutes prior to bedtime and limit recreational screen time to under 3.5 hours/day.",
            domain: "Digital Overload / Sleep Quality",
            milestone: "Reduce average screen time by 1.5 hours daily"
        }
    ]
};

function getRecommendations(score, domainProfile) {
    const recs = [];
    const depr = domainProfile["Depression"] || 0;
    const anx  = domainProfile["Anxiety"] || 0;
    const slp  = domainProfile["Sleep Disruption"] || 0;

    if (depr >= anx && depr >= 30) {
        recs.push(...CBT_DATA.depression);
    } else if (anx > depr && anx >= 30) {
        recs.push(...CBT_DATA.anxiety);
    } else {
        recs.push(CBT_DATA.anxiety[0], CBT_DATA.depression[0]);
    }

    if (slp >= 35 || domainProfile["Lifestyle & Behavioral Risk"] >= 40) {
        recs.push(CBT_DATA.sleep[0]);
        recs.push(CBT_DATA.lifestyle[0]);
    }

    return recs.slice(0, 4);
}

function classifyRisk(score) {
    if (score < 20) return { label: "Tier 1: Minimal Risk", badge: "badge-minimal", color: "#10B981", desc: "Symptoms within healthy normative bounds. Continued wellness monitoring." };
    if (score < 40) return { label: "Tier 2: Mild Risk", badge: "badge-mild", color: "#3B82F6", desc: "Mild sub-clinical symptoms. Digital psychoeducation and lifestyle self-care recommended." };
    if (score < 60) return { label: "Tier 3: Moderate Risk", badge: "badge-moderate", color: "#F59E0B", desc: "Moderate symptoms. Active clinician check-in and targeted CBT protocol recommended." };
    if (score < 80) return { label: "Tier 4: High Risk", badge: "badge-high", color: "#F97316", desc: "High clinical risk. Prioritize clinical safety review and formal psychiatric assessment." };
    return { label: "Tier 5: Severe Risk", badge: "badge-severe", color: "#EF4444", desc: "Critical clinical severity. Immediate clinical escalation and emergency safety planning required." };
}

function generateFHIRJSON(patientName, score, ci, domainProfile, recs) {
    return {
        resourceType: "DiagnosticReport",
        id: "shri-assessment-" + Date.now(),
        status: "final",
        category: [{ coding: [{ system: "http://terminology.hl7.org/CodeSystem/v2-0074", code: "MB", display: "Behavioral Health" }] }],
        code: { coding: [{ system: "http://loinc.org", code: "96767-9", display: "Mental Health Risk Assessment" }] },
        subject: { display: patientName },
        effectiveDateTime: new Date().toISOString(),
        issued: new Date().toISOString(),
        conclusion: `Smart Health Risk Score: ${score}/100 with 95% CI [${ci[0]} - ${ci[1]}]. Clinical Tier: ${classifyRisk(score).label}`,
        domainScores: domainProfile,
        recommendedActions: recs.map(r => r.title)
    };
}
